src/test/java/com/neo4j/docker/utils/Neo4jVersionTest.java

```diff
@@ -0,0 +1,193 @@
+                     String.format( "Didn't detect that %s is older than %s using isAtLeastVersion", older, newer ) );
+
+        Assertions.assertFalse( newer.isEqual(older),
+                                String.format( "%s incorrectly is equal to %s", newer, older ) );
+        Assertions.assertNotEquals( newer, older, String.format( "%s incorrectly is equal to %s", newer, older ) );
+    }
+
+    @Test
+    public void testIsOlderThan_sameReleaseReturnsFalse()
+    {
+        Neo4jVersion version = new Neo4jVersion( 3, 5, 1 );
+
+        Assertions.assertFalse( version.isOlderThan( version ) ,
+                     "A release should not be older than itself" );
+        Assertions.assertFalse( version.isNewerThan( version ) ,
+                     "A release should not be newer than itself" );
+    }
+
+    @Test
+    void testIsCalVer_2024()
+    {
+        Neo4jVersion version = new Neo4jVersion(2024, 9, 0);
+        Assertions.assertTrue(version.isCalVer(), version+" should be flagged as a calendar version");
+    }
+
+    @Test
+    void testIsCalVer_2025()
+    {
+        Neo4jVersion version = new Neo4jVersion(2025, 1, 0);
+        Assertions.assertTrue(version.isCalVer(), version+" should be flagged as a calendar version");
+    }
+
+    @Test
+    void testIsCalVer_5x()
+    {
+        Neo4jVersion version = new Neo4jVersion(5, 1, 0);
+        Assertions.assertFalse(version.isCalVer(), version+" should NOT be flagged as a calendar version");
+    }
+
+    @Test
+    public void testFromVersionString_releaseFormat()
+    {
+        Neo4jVersion version = Neo4jVersion.fromVersionString( "4.4.7" );
+        Assertions.assertEquals( 4, version.major, "Did not parse major number from " + version );
+        Assertions.assertEquals( 4, version.minor, "Did not parse minor number from " + version );
+        Assertions.assertEquals( 7, version.patch, "Did not parse patch number from " + version );
+    }
+
+    @Test
+    public void testFromVersionString_BSPFormat()
+    {
+        Neo4jVersion version = Neo4jVersion.fromVersionString( "4.4.7-12345" );
+        Assertions.assertEquals( 4, version.major, "Did not parse major number from " + version );
+        Assertions.assertEquals( 4, version.minor, "Did not parse minor number from " + version );
+        Assertions.assertEquals( 7, version.patch, "Did not parse patch number from " + version );
+        Assertions.assertEquals( "-12345", version.label, "Did not build number from " + version );
+    }
+
+    @Test
+    public void testFromVersionString_calver_releaseFormat()
+    {
+        Neo4jVersion version = Neo4jVersion.fromVersionString( "2024.10.0" );
+        Assertions.assertEquals( 2024, version.major, "Did not parse major number from " + version );
+        Assertions.assertEquals( 10, version.minor, "Did not parse minor number from " + version );
+        Assertions.assertEquals( 0, version.patch, "Did not parse patch number from " + version );
```

Comment:
Minor comment: Major, minor and patch maybe don't make sense anymore? But calling them firstDigit, secondDigit, patch is ugly so 🤷 .

Reply:
yeah. There's no good solution since 5.x and 4.4 are expected to still use major/minor.